waspc/src/Wasp/Generator/DbGenerator/Jobs.hs

```diff
@@ -46,31 +47,38 @@ migrateDev projectDir migrateArgs =
     scriptArgs =
       if System.Info.os == "darwin"
         then -- NOTE(martin): On MacOS, command that `script` should execute is treated as multiple arguments.
-          ["-Fq", "/dev/null"] ++ prismaMigrateCmd
+          ["-Fq", "/dev/null"] ++ Shell.getShellArgValues prismaMigrateCmd
         else -- NOTE(martin): On Linux, command that `script` should execute is treated as one argument.
-          ["-feqc", unwords prismaMigrateCmd, "/dev/null"]
+        -- NOTE(miho): Since we are passing the arguments as a single string, we need to wrap the arguments
+        --  which might contain spaces in them. We are using Shell.Quoted to know which
+        --   arguments should be wrapped in quotes and which shouldn't.
+        --   Specifically, we are using `showShellArgs` to wrap file paths in quotes. Other arguments
+        --   are not wrapped in quotes since we know their values and they don't contain spaces.
```

Comment:
Where are we passing arguments as a single string?
Aha, now I realized there is a comment above saying that on Linux, command that script should execute is treated as one argument.